src/modules/avformat/consumer_avformat.c

```diff
@@ -1776,6 +1762,22 @@ static void *consumer_thread( void *arg )
 							converted_avframe->data, converted_avframe->linesize);
 						sws_freeContext( context );
 
+#else
+						video_avframe.interlaced_frame = !mlt_properties_get_int( frame_properties, "progressive" );
+						video_avframe.format = avformat_map_pixfmt_mlt2av( img_fmt );
```

Comment:
Behavior of avformat_map_pixfmt_mlt2av() differs slightly than pick_pix_fmt(), but it might be OK. I do not think anyone tries to use mlt_image_opengl or mlt_image_glsl_texture with the avformat consumer,